src/layers/tanh.rs

```diff
@@ -0,0 +1,34 @@
+use crate::layers::layer::Layer;
+use crate::tensor::Tensor;
+
+pub struct Tanh;
+
+impl Layer for Tanh {
+    fn get_info(&self) -> String {
+        format!("Tanh Layer")
+    }
+
+    fn forward(&self, input: &Tensor) -> Tensor {
+        input.map(|x| Activation::tanh(x))
+    }
+
+    fn backward(&mut self, input: &Tensor, grad_output: Tensor) -> Tensor {
+        let tanh_grad = input.map(|x| Activation::tanh_prime(x));
+        println!("grad tanh: \n{}", grad_output.mult_el(&tanh_grad));
+        grad_output.mult_el(&tanh_grad)
+    }
+}
+
+struct Activation;
```

Comment:
Why a new struct ? You can put these 2 functions inside the Tanh struct :smile: and `impl Tanh` instead.

Reply:
My first attempt was to put every function in `impl Tanh`, but I couldn't find any way to call `tanh` and `tanh_prime` from `forward` and `backward` methods...

Reply:
Solved by : e062462